File: Hrms.Core/Models/JobApplication/InterviewModel.cs
using Hrms.Core.Utilities;

namespace Hrms.Core.Models.JobApplication
{
    public class InterviewModel
    {
        public int Id { get; set; }
        public Constants.InterviewType InterviewType { get; set; }
        public Constants.InterviewMode InterviewMode { get; set; }
        public int CandidateId { get; set; }
        public DateTime ScheduleDate { get; set; }
        public DateTime? InterviewDate { get; set; }
        public TimeSpan? ScheduleTime { get; set; }
        public int? Rating { get; set; }
        public bool? EligibleForNextRound { get; set; }
        public string InterviewerName { get; set; }
        public string Remark { get; set; }
        public int InterviewerId { get; set; }
        public string LegalName { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public Constants.RecordStatus Status { get;set; }
        public FileDetailModel DocumentDetails { get; set; }
    }
}

File: Hrms.Web/src/app/shared/models/interview.model.ts
import { FileDetailModel } from "src/app/employee/job-recruitment/model";

export class InterviewModel {
    id: number;
    interviewMode: number;
    interviewType: number;
    interviewDate: string;
    scheduleDate: string;
    scheduleTime: string;
    interviewerId: number;
    candidateId: number;
    rating: number;
    firstName:string;
    lastName:string;
    eligibleForNextRound: boolean;
    remark: string;
    legalName: string;
    email: string;
    phone:string;
    interviewerName:string;
    courseName:string;
    stream:string;
    passingYear: number;
    shortlistedDate: string;
    status: string;
    documentDetails: FileDetailModel;

    constructor(){
        this.eligibleForNextRound = null;
        this.documentDetails = new FileDetailModel();
    }
}